src/dash-table/components/Table/index.tsx

```diff
@@ -9,50 +9,64 @@ import derivedPaginator from 'dash-table/derived/paginator';
 import derivedSelectedRows from 'dash-table/derived/selectedRows';
 import derivedViewportData from 'dash-table/derived/data/viewport';
 import derivedVirtualData from 'dash-table/derived/data/virtual';
+import derivedVirtualizedData from 'dash-table/derived/data/virtualized';
 import derivedVisibleColumns from 'dash-table/derived/column/visible';
 
 import {
     ControlledTableProps,
     PropsWithDefaultsAndDerived,
-    SetProps
+    SetProps,
+    IState,
+    StandaloneState,
+    PropsWithDefaults
 } from './props';
 
 import 'react-select/dist/react-select.css';
 import './Table.less';
 import './Dropdown.css';
+import { isEqual } from 'core/comparer';
+/*#if DEV*/
+import Logger from 'core/Logger';
+/*#endif*/
 
 const DERIVED_REGEX = /^derived_/;
 
-export default class Table extends Component<PropsWithDefaultsAndDerived> {
+export default class Table extends Component<PropsWithDefaultsAndDerived, StandaloneState> {
```

Comment:
Improving internal typing for standalone case by defining the State type -- the controlled table's state has been entirely moved back to the table now and modifies it through props.setState

This is necessary because we need to hook up to the scroll event and to eval various element sizes to make virtualization work correctly.